### PR TITLE
Task 3

### PR DESCRIPTION
**Pull request by RuisuLee**
### Done:
- Service is done
- FE part (CloudFront: https://d3rz6g0e2mzffy.cloudfront.net/)

### Additional scope:
- Async/await
- ES6 modules
- Webpack
- Lambda handlers are separated
- Main error scenarios are handled
- SWAGGER documentation: https://app.swaggerhub.com/apis/Ruisu_Lee/Products/1.0.0

### Link to product-service API:
https://mxwvyv7tnl.execute-api.eu-west-1.amazonaws.com/dev/products
https://mxwvyv7tnl.execute-api.eu-west-1.amazonaws.com/dev/products/7567ec4b-b10c-48c5-9345-fc73c48a80a2

### Link to FE MR:
https://github.com/RuisuLee/nodejs-aws-fe/pull/2

### CODE REVIEW

File: product-service/handlers/getProductsById.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+export const getProductsById: APIGatewayProxyHandler = async (event, _context) => {
+  const productsService = new ProductsService();
+  const { productId } = event.pathParameters;
+  try {
+    const product = await productsService.getProductById(productId);
+    return {
+      statusCode: 200,
+      body: JSON.stringify(product),
+      headers: {
+        'Access-Control-Allow-Origin': '*',
```
Comment (MaxGaranin) [inline product-service/handlers/getProductsById.ts]: These headers can be moved to constants, they repeating in some code parts.

File: product-service/services/products.service.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+export class ProductsService {
+  constructor() {}
+
+  public getAllProducts() {
+    return products;
+  }
+
+  public getProductById(id) {
+    return new Promise((resolve, reject) => {
+      const productById = products.filter(product => { return product.id === id })[0];
```
Comment (MaxGaranin) [inline product-service/services/products.service.ts]: I suppose, method 'find' will be more suitable.

**Review by MaxGaranin — APPROVED** [review]

File: product-service/handlers/getProductsById.ts (hunk 1 above)
Comment (RuisuLee) [reply product-service/handlers/getProductsById.ts]: Thank you! It's a good point.

File: product-service/services/products.service.ts (hunk 2 above)
Comment (RuisuLee) [reply product-service/services/products.service.ts]: Thanks for suggestion, I'll rework this.

**Pull request merged**